src/artemis/tests/test_fgs_communicator.py

```diff
@@ -0,0 +1,265 @@
+    assert communicator.params == FullParameters()
+
+    communicator.results = "some position to move to"
+    communicator.reset(FullParameters())
+    assert communicator.results is None
+
+
+@patch("artemis.fgs_communicator.run_start")
+@patch("artemis.fgs_communicator.run_end")
+@patch("artemis.fgs_communicator.wait_for_result")
+@patch("artemis.fgs_communicator.StoreInIspyb3D.store_grid_scan")
+@patch("artemis.fgs_communicator.StoreInIspyb3D.get_current_time_string")
+@patch(
+    "artemis.fgs_communicator.StoreInIspyb3D.update_grid_scan_with_end_time_and_status"
+)
+def test_run_gridscan_zocalo_calls(
+    mock_ispyb_update_time_and_status: MagicMock,
+    mock_ispyb_get_time: MagicMock,
+    mock_ispyb_store_grid_scan: MagicMock,
+    wait_for_result: MagicMock,
+    run_end,
+    run_start,
+):
+
+    dc_ids = [1, 2]
+    dcg_id = 4
+
+    mock_ispyb_store_grid_scan.return_value = [dc_ids, None, dcg_id]
+    mock_ispyb_get_time.return_value = DUMMY_TIME_STRING
+    mock_ispyb_update_time_and_status.return_value = None
+
+    params = FullParameters()
+    params.detector_params.prefix += str(time.time())
+    communicator = fgs_communicator.FGSCommunicator()
+    communicator.reset(params)
+    communicator.start(test_start_document)
+    communicator.descriptor(test_descriptor_document)
+    communicator.event(test_event_document)
+    communicator.stop(test_stop_document)
+
+    run_start.assert_has_calls(call(x) for x in dc_ids)
+    assert run_start.call_count == len(dc_ids)
+
+    run_end.assert_has_calls(call(x) for x in dc_ids)
+    assert run_end.call_count == len(dc_ids)
+
+    wait_for_result.assert_called_once_with(dcg_id)
+
+
+@pytest.fixture
+def dummy_3d_gridscan_args():
+    params = FullParameters()
+    params.grid_scan_params.z_steps = 2
+
+    FakeFGSComposite = make_fake_device(FGSComposite)
+    fgs_composite: FGSComposite = FakeFGSComposite(name="fgs", insertion_prefix="")
+
+    FakeEiger = make_fake_device(EigerDetector)
+    eiger: EigerDetector = FakeEiger(
+        detector_params=params.detector_params, name="eiger"
+    )
+    communicator = fgs_communicator.FGSCommunicator()
+    communicator.xray_centre_motor_position = Point3D(1, 2, 3)
+
+    return fgs_composite, eiger, params, communicator
+
+
+@patch("artemis.fgs_communicator.run_start")
+@patch("artemis.fgs_communicator.run_end")
+@patch("artemis.fgs_communicator.wait_for_result")
+@patch("artemis.fgs_communicator.StoreInIspyb3D.store_grid_scan")
+@patch("artemis.fgs_communicator.StoreInIspyb3D.get_current_time_string")
+@patch(
+    "artemis.fgs_communicator.StoreInIspyb3D.update_grid_scan_with_end_time_and_status"
+)
+def test_fgs_failing_results_in_bad_run_status_in_ispyb(
+    mock_ispyb_update_time_and_status: MagicMock,
+    mock_ispyb_get_time: MagicMock,
+    mock_ispyb_store_grid_scan: MagicMock,
+    wait_for_result: MagicMock,
+    run_end: MagicMock,
+    run_start: MagicMock,
+    dummy_3d_gridscan_args,
```

Comment:
Nit: Do you need `dummy_3d_gridscans_args`, I don't think you're really using it? If you don't you can remove the fixture